mp/src/game/shared/sdk/weapon_sdkbase.cpp

```diff
@@ -449,6 +449,9 @@ void CWeaponSDKBase::StartSwing(bool bIsSecondary, bool bIsStockAttack)
 	if ( !pOwner )
 		return;
 
+	if (pOwner->GetPunchesThrown() > 0)
+		return;
+
```

Comment:
If you want to make punches semi auto, I don't think you have to track punches thrown, I think you can just check whether the key was pressed as opposed to down.

Reply:
That information os available several function calls further up, in each weapon's ItemPostFrame, but not here, where all melee converges :)

Reply:
btw, like i said, this probably needs some further touches, please don't merge yet

Reply:
Still not sure whether to make melee semi-auto or keep it full-auto.
I've seen people abuse it by just typing +attack2 in the console and then leaving it on to get that extra damage in case they're near someone, but otherwise just using guns.
But then there are lag issues that basically require you to spam to hit.

What's your opinion?
Since I already have a count, maybe allow bursts of 3 (ish) punches?

Reply:
My opinion is disable shooting for a time after you melee if you want to fix the +attack2 problem.

Reply:
I think that'd be too much of a melee nerf and in the long run it'd mostly affect less experienced players anyway, since the pros would get used to the new way quickly.
Also, that'd be one more thing that can have bugs, whereas that melee burst system was just a copypasta of the existing burst system.